app/src/main/java/fr/free/nrw/commons/upload/structure/depicts/DepictModel.java

```diff
@@ -0,0 +1,174 @@
+package fr.free.nrw.commons.upload.structure.depicts;
+
+import android.text.TextUtils;
+
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.Comparator;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+
+import fr.free.nrw.commons.category.CategoryClient;
+import fr.free.nrw.commons.kvstore.JsonKvStore;
+import fr.free.nrw.commons.upload.depicts.DepictsInterface;
+import fr.free.nrw.commons.utils.StringSortingUtils;
+import io.reactivex.Observable;
+import io.reactivex.disposables.CompositeDisposable;
+import timber.log.Timber;
+
+public class DepictModel {
+    private static final int SEARCH_DEPICTS_LIMIT = 25;
+    private final DepictDao depictDao;
+    private final DepictsInterface depictsInterface;
+    private final JsonKvStore directKvStore;
+    @Inject
+    GpsDepictsModel gpsDepictsModel;
+    @Inject
+    CategoryClient categoryClient;
+
+    private List<DepictedItem> selectedDepictedItems;
+    private HashMap<String, ArrayList<String>> depictsCache;
+    private CompositeDisposable compositeDisposable = new CompositeDisposable();
+
+    @Inject
+    public DepictModel(DepictDao depictDao, @Named("default_preferences") JsonKvStore directKvStore, DepictsInterface depictsInterface) {
+        this.depictDao = depictDao;
+        this.directKvStore = directKvStore;
+        this.depictsInterface = depictsInterface;
+        this.depictsCache = new HashMap<>();
+        this.selectedDepictedItems = new ArrayList<>();
+    }
+
+    public Comparator<DepictedItem> sortBySimilarity(final String filter) {
+        Comparator<String> stringSimilarityComparator = StringSortingUtils.sortBySimilarity(filter);
+        return (firstItem, secondItem) -> stringSimilarityComparator
+                .compare(firstItem.getDepictsLabel(), secondItem.getDescription());
+    }
+
+    public boolean containsYear(String item) {
```

Comment:
This could be a candidate for unit test

Reply:
Ashish But currently this function isn't being used anywhere in the code

Reply:
You mean this is not used anywhere, if yes, we can remove it